File: Back-end/HealthMindBackend/HealthMindBackend.Application/Services/PlanoSaudeService.cs
using AutoMapper;
using HealthMindBackend.Application.CoberturasPlanos.Commands;
using HealthMindBackend.Application.DTOs;
using HealthMindBackend.Application.Interfaces;
using HealthMindBackend.Application.PlanosSaude.Commands;
using HealthMindBackend.Application.PlanosSaude.Queries;
using MediatR;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HealthMindBackend.Application.Services
{
    public class PlanoSaudeService : IPlanoSaudeService
    {
        private readonly IMapper _mapper;
        private readonly IMediator _mediator;

        public PlanoSaudeService(IMapper mapper, IMediator mediator)
        {
            _mapper = mapper;
            _mediator = mediator;
        }

        public async Task AtualizarCoberturaPlano(String planoSaudeId, CoberturaPlanoDTO coberturaPlanoDto)
        {
            var coberturaPlanoUpdateCommand = _mapper.Map<CoberturaPlanoUpdateCommand>(coberturaPlanoDto);
            coberturaPlanoUpdateCommand.PlanoSaudeId = planoSaudeId;
            await _mediator.Send(coberturaPlanoUpdateCommand);
        }

        public async Task AtualizarPlanoSaude(PlanoSaudeDTO planoSaudeDto)
        {
            var planoSaudeUpdateCommand = _mapper.Map<PlanoSaudeUpdateCommand>(planoSaudeDto);
            await _mediator.Send(planoSaudeUpdateCommand);
        }

        public async Task<IEnumerable<PlanoSaudeDTO>> GetAllPlanosSaude()
        {
            var getAllPlanosSaude = new GetAllPlanosSaudeQuery();
            var result = await _mediator.Send(getAllPlanosSaude);
            return _mapper.Map<IEnumerable<PlanoSaudeDTO>>(result);
        }

        public async Task RegistrarCoberturaPlano(String planoSaudeId, CoberturaPlanoDTO coberturaPlanoDto)
        {
            var coberturaPlanoCreateCommand = _mapper.Map<CoberturaPlanoCreateCommand>(coberturaPlanoDto);
            coberturaPlanoCreateCommand.PlanoSaudeId = planoSaudeId;
            await _mediator.Send(coberturaPlanoCreateCommand);
        }

        public async Task RegistrarPlanoSaude(PlanoSaudeDTO planoSaudeDto)
        {
            var planoSaudeCreateCommand = _mapper.Map<PlanoSaudeCreateCommand>(planoSaudeDto);
            await _mediator.Send(planoSaudeCreateCommand);
        }

        public async Task RemoverCoberturaPlano(String planoSaudeId, String especialidade)
        {
            var coberturaPlanoDeleteCommand = new CoberturaPlanoDeleteCommand(planoSaudeId, especialidade);
            await _mediator.Send(coberturaPlanoDeleteCommand);
        }
    }
}

File: Front-end/health-mind-system-frontend/src/shared/services/plano-saude.service.ts
import { api } from "./api";
import { PlanoSaudeDTO } from "../types/dtos/PlanoSaude.dto";
import { CoberturaPlanoDTO } from "../types/dtos/CoberturaPlano.dto";


const controller = "planoSaude";

export async function getAllPlanosSaude(): Promise<PlanoSaudeDTO[]> {
    const response = await api.get<PlanoSaudeDTO[]>(controller);
    return response.data;
}

export async function registrarPlanoSaude(planoSaudeDto: PlanoSaudeDTO): Promise<PlanoSaudeDTO> {
    const response = await api.post(controller, planoSaudeDto);
    return response.data;
}

export async function atualizarPlanoSaude(planoSaudeId: string, planoSaudeDto: PlanoSaudeDTO): Promise<PlanoSaudeDTO> {
    const response = await api.put(`${controller}/${planoSaudeId}`, planoSaudeDto);
    return response.data;
}

export async function registrarCoberturaPlano(planoSaudeId: string, coberturaPlanoDto: CoberturaPlanoDTO): Promise<CoberturaPlanoDTO> {
    const response = await api.post(`${controller}/${planoSaudeId}/coberturas-plano`, coberturaPlanoDto);
    return response.data;
}

export async function atualizarCoberturaPlano(planoSaudeId: string, especialidade: string, coberturaPlanoDto: CoberturaPlanoDTO): Promise<CoberturaPlanoDTO> {
    const response = await api.put(`${controller}/${planoSaudeId}/coberturas-plano/${especialidade}`, coberturaPlanoDto);
    return response.data;
}

export async function removerCoberturaPlano(planoSaudeId: string, especialidade: string): Promise<void> {
    const response = await api.delete(`${controller}/${planoSaudeId}/coberturas-plano/${especialidade}`)
    return response.data;
}